File: Linnworks/src/net/ExternalApplications.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;

namespace LinnworksAPI
{
    public static class ExternalApplicationsMethods
    {
        public static void RevokeToken(Guid token, String ApiToken, String ApiServer)
        {
            Factory.GetResponse("ExternalApplications/RevokeToken", "token=" + token + "", ApiToken, ApiServer);
        }

        public static Boolean LeaveReview(Guid applicationId, Int32 rating, String comment, String ApiToken, String ApiServer)
        {
            return Newtonsoft.Json.JsonConvert.DeserializeObject<Boolean>(Factory.GetResponse("ExternalApplications/LeaveReview", "applicationId=" + applicationId + "&rating=" + rating + "&comment=" + comment + "", ApiToken, ApiServer), new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
        }

        public static GetApplicationResponse GetApplication(Guid applicationId, Boolean loadReviews, String ApiToken, String ApiServer)
        {
            return Newtonsoft.Json.JsonConvert.DeserializeObject<GetApplicationResponse>(Factory.GetResponse("ExternalApplications/GetApplication", "applicationId=" + applicationId + "&loadReviews=" + loadReviews + "", ApiToken, ApiServer), new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
        }

        public static List<DeveloperApplication> SearchApplications(String searchString, Int32 count, String ApiToken, String ApiServer)
        {
            return Newtonsoft.Json.JsonConvert.DeserializeObject<List<DeveloperApplication>>(Factory.GetResponse("ExternalApplications/SearchApplications", "searchString=" + searchString + "&count=" + count + "", ApiToken, ApiServer), new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
        }

        public static List<ApplicationToken> GetTokens(String ApiToken, String ApiServer)
        {
            return Newtonsoft.Json.JsonConvert.DeserializeObject<List<ApplicationToken>>(Factory.GetResponse("ExternalApplications/GetTokens", "", ApiToken, ApiServer), new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
        }

        public static List<ApplicationCategory> GetCategories(String ApiToken, String ApiServer)
        {
            return Newtonsoft.Json.JsonConvert.DeserializeObject<List<ApplicationCategory>>(Factory.GetResponse("ExternalApplications/GetCategories", "", ApiToken, ApiServer), new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
        }
    }
}

File: Linnworks/src/javascript/ExternalApplications.js
var ExternalApplications =
{
	// http://apidoc.linnworks.net/#/ExternalApplications-RevokeToken
	RevokeToken: function(token,token, server)
	{
		return Factory.GetResponse("ExternalApplications/RevokeToken", token, server, "token=" + token +"");
	},
	// http://apidoc.linnworks.net/#/ExternalApplications-UninstallApplication
	UninstallApplication: function(InstallationId,token, server)
	{
		return Factory.GetResponse("ExternalApplications/UninstallApplication", token, server, "InstallationId=" + InstallationId +"");
	},
	// http://apidoc.linnworks.net/#/ExternalApplications-LeaveReview
	LeaveReview: function(applicationId,rating,comment,token, server)
	{
		return Factory.GetResponse("ExternalApplications/LeaveReview", token, server, "applicationId=" + applicationId + "&rating=" + rating + "&comment=" + comment +"");
	},
	// http://apidoc.linnworks.net/#/ExternalApplications-GetTemporaryToken
	GetTemporaryToken: function(ApplicationInstallationId)
	{
		return Factory.GetResponse("ExternalApplications/GetTemporaryToken", "", "", "ApplicationInstallationId=" + ApplicationInstallationId +"");
	},
	// http://apidoc.linnworks.net/#/ExternalApplications-GetApplication
	GetApplication: function(applicationId,loadReviews,token, server)
	{
		return Factory.GetResponse("ExternalApplications/GetApplication", token, server, "applicationId=" + applicationId + "&loadReviews=" + loadReviews +"");
	},
	// http://apidoc.linnworks.net/#/ExternalApplications-SearchApplications
	SearchApplications: function(searchString,count,token, server)
	{
		return Factory.GetResponse("ExternalApplications/SearchApplications", token, server, "searchString=" + searchString + "&count=" + count +"");
	},
	// http://apidoc.linnworks.net/#/ExternalApplications-GetTokens
	GetTokens: function(token, server)
	{
		return Factory.GetResponse("ExternalApplications/GetTokens", token, server, "");
	},
	// http://apidoc.linnworks.net/#/ExternalApplications-GetInstalled
	GetInstalled: function(token, server)
	{
		return Factory.GetResponse("ExternalApplications/GetInstalled", token, server, "");
	},
	// http://apidoc.linnworks.net/#/ExternalApplications-DownloadLinnworksDesktopClient
	DownloadLinnworksDesktopClient: function(token, server)
	{
		return Factory.GetResponse("ExternalApplications/DownloadLinnworksDesktopClient", token, server, "");
	},
};
